### PR TITLE
HBASE-22550 Do not use Threads.newDaemonThreadFactory in ConnectionUt…

### PR DESCRIPTION
**Pull request by Apache9**
…ils.getThreadPool

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 168 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 1 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ HBASE-21512 Compile Tests _ |
| +1 | mvninstall | 260 | HBASE-21512 passed |
| +1 | compile | 23 | HBASE-21512 passed |
| +1 | checkstyle | 45 | HBASE-21512 passed |
| +1 | shadedjars | 268 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 62 | HBASE-21512 passed |
| +1 | javadoc | 21 | HBASE-21512 passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 235 | the patch passed |
| +1 | compile | 23 | the patch passed |
| +1 | javac | 23 | the patch passed |
| +1 | checkstyle | 26 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 262 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 739 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | findbugs | 65 | the patch passed |
| +1 | javadoc | 22 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 108 | hbase-client in the patch passed. |
| +1 | asflicense | 9 | The patch does not generate ASF License warnings. |
| | | 2642 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-291/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/291 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 6f7ff550ad95 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | HBASE-21512 / 3229554fdf |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-291/1/testReport/ |
| Max. process+thread count | 293 (vs. ulimit of 10000) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-291/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionOverAsyncConnection.java [hunk 1]
```diff
@@ -134,13 +139,25 @@ public boolean isClosed() {
     return conn.isClosed();
   }
 
+  private ThreadPoolExecutor createThreadPool() {
+    Configuration conf = conn.getConfiguration();
+    int threads = conf.getInt("hbase.hconnection.threads.max", 256);
```
Comment (saintstack) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionOverAsyncConnection.java]: Pool could spin up to 256 threads? The old client used to do this but I was hoping new async client would leave behind old days and new async client would run w/ a few threads only? It'd run smarter.

Comment (Apache9) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionOverAsyncConnection.java]: This is just for the legacy coprocessor methods, if you do not call coprocessor related methods then there will be no thread pool. And the coprocessor methods have been deprecated and will be removed in 4.0.0, users are expected to call the ones in async client instead.

Comment (saintstack) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionOverAsyncConnection.java]: Ok. Thanks.

**Pull request merged**